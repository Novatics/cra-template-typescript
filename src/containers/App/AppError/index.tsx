import React from 'react';
import i18n from 'common/i18n/config';
import { ErrorBoundary, FallbackProps } from 'react-error-boundary';

export interface AppErrorProps {
  children: JSX.Element;
}

const AppError: React.FC<AppErrorProps> = (props: AppErrorProps) => {
  function ErrorFallback({ error, resetErrorBoundary }: FallbackProps) {
    return (
      <div>
        <h1>{i18n.t('error.boundary.somethings-wrong')}</h1>
        <pre>{error.message}</pre>
        <button onClick={resetErrorBoundary}>Try again</button>
      </div>
    );
  }

  return <ErrorBoundary FallbackComponent={ErrorFallback}>{props.children}</ErrorBoundary>;
};

export default AppError;
