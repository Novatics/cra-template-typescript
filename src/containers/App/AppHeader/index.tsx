import React from 'react';
import { Link as LinkRouter } from 'react-router-dom';
import { ReactComponent as Logo } from 'assets/images/logo.svg';
import { Container } from './styles';

const AppHeader = () => {
  return (
    <Container>
      <LinkRouter to="/">
        <Logo width="160" />
      </LinkRouter>
    </Container>
  );
};

export default AppHeader;
