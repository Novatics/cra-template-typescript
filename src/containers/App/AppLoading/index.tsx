import React from 'react';
import './styles.css';

const AppLoading: React.FC = () => {
  return (
    <div className="container">
      <div className="spinner">
        <div className="cube1"></div>
        <div className="cube2"></div>
      </div>
    </div>
  );
};

export default AppLoading;
