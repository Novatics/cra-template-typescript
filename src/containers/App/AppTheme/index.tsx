import React from 'react';
import { ThemeProvider } from 'styled-components';
import GlobalStyle from 'common/globalStyles';

const theme = {
  fonts: ['Work Sans', 'sans-serif', 'Roboto'],
  fontSizes: {
    small: '1em',
    medium: '2em',
    large: '2.2em',
    xlarge: '2.5em'
  },
  spaces: {
    xsmall: '0.2em',
    small: '0.5em',
    normal: '0.7em',
    medium: '1em'
  },
  colors: {
    text: {
      primary: '#eeeeee',
      secondary: '#b9b9b9'
    },
    bg: {
      darker: '#371c9d',
      dark: '#371c9d'
    },
    border: {
      dark: '#4b4a4a'
    }
  }
};

export interface AppThemeProps {
  children: JSX.Element;
}

const AppTheme: React.FC<AppThemeProps> = ({ children }: AppThemeProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};

export default AppTheme;
