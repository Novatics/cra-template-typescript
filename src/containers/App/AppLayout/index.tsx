import React from 'react';
import AppHeader from '../AppHeader';

export interface AppLayoutProps {
  children: JSX.Element;
}

const AppLayout: React.FC<AppLayoutProps> = (props: AppLayoutProps) => {
  return (
    <>
      <AppHeader />
      {props.children}
    </>
  );
};

export default AppLayout;
