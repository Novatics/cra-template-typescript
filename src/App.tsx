import React, { Suspense } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import AppError from 'containers/App/AppError';
import AppLayout from 'containers/App/AppLayout';
import AppLoading from 'containers/App/AppLoading';
import AppTheme from 'containers/App/AppTheme';
import routes from 'routes';
import 'common/i18n/config';

const App: React.FC = () => {
  return (
    <AppTheme>
      <AppError>
        <Suspense fallback={<AppLoading />}>
          <BrowserRouter>
            <AppLayout>
              <Switch>
                {routes.map(route => (
                  <Route
                    key={route.path}
                    path={route.path}
                    exact={route.exact}
                    component={route.component}
                  />
                ))}
              </Switch>
            </AppLayout>
          </BrowserRouter>
        </Suspense>
      </AppError>
    </AppTheme>
  );
};

export default App;
