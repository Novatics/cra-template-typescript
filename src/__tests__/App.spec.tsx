import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
