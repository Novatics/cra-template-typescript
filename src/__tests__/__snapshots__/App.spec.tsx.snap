// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="Ps-MuiPaper-root Ps-MuiAppBar-root Ps-MuiAppBar-positionFixed Ps-MuiAppBar-colorDefault mui-fixed Ps-MuiPaper-elevation1"
    style="display: none;"
  >
    <div
      class="Ps-MuiToolbar-root Ps-MuiToolbar-regular Ps-MuiToolbar-gutters"
    >
      <a
        href="/"
      />
    </div>
  </header>
  <div
    class="Ps-MuiBox-root Ps-MuiBox-root-1"
    style="display: none;"
  />
  <div
    class="Ps-MuiBox-root Ps-MuiBox-root-2"
  >
    <div
      class="Ps-MuiCircularProgress-root Ps-MuiCircularProgress-colorPrimary Ps-MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="Ps-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="Ps-MuiCircularProgress-circle Ps-MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
